src/settings.js

```diff
@@ -1597,9 +1597,13 @@ var USE_SQLITE3 = false;
 // [compile+link] - affects user code at compile and system libraries at link.
 var SHARED_MEMORY = false;
 
-// If true, enables support for Wasm Workers. Wasm Workers enable applications
+// If 1, enables support for Wasm Workers. Wasm Workers enable applications
 // to create threads using a lightweight web-specific API that builds on top
-// of Wasm SharedArrayBuffer + Atomics API.
+// of Wasm SharedArrayBuffer + Atomics API. When enabled, a new build output
+// file a.ww.js will be generated to bootstrap the Wasm Worker JS contexts.
+// If 2, enables support for Wasm Workers, but without using a separate a.ww.js
+// file on the side. This can simplify deployment of builds, but will have a
+// downside that the generated build will no longer be csp-eval compliant.
```

Comment:
This sounds like just embedding the worker JS boostrap file inside the main JS file.  However, in the other issue regarding this it sounded like i also related the generated html file.   Am I misunderstanding something?

Reply:
Yeah, `-sWASM_WORKERS=2` mode embeds the .ww.js file into the main .js file.

Commented in https://github.com/emscripten-core/emscripten/issues/17547#issuecomment-2022263978 for more overview, maybe that helps.